common/src/main/scala/org/genivi/sota/http/AuthNamespaceDirectives.scala

```diff
@@ -1,54 +1,76 @@
 package org.genivi.sota.http
 
 import akka.http.scaladsl.model.headers.OAuth2BearerToken
-import akka.http.scaladsl.server.{AuthorizationFailedRejection, Directive1, Directives, Rejection}
+import akka.http.scaladsl.server.{AuthorizationFailedRejection, Directive0, Directive1, Directives, Rejection}
+import akka.http.scaladsl.server.Directives._
 import cats.data.Xor
+import com.advancedtelematic.akka.http.jwt.InvalidScopeRejection
 import com.advancedtelematic.jws.CompactSerialization
-import com.advancedtelematic.jwt.{JsonWebToken, Subject}
+import com.advancedtelematic.jwt.{JsonWebToken, Scope, Subject}
 import io.circe.parser._
 import org.genivi.sota.data.Namespace._
 import eu.timepit.refined.refineV
 import io.circe.Decoder
 import org.genivi.sota.data.Namespace
 
+case class AuthedNamespaceScope(namespace: Namespace, scope: Scope, owned: Boolean) {
+  type ScopeItem = String
+
+  def hasScope(sc: ScopeItem) : Boolean = owned || scope.underlying.contains(sc)
+
+  def hasScopeReadonly(sc: ScopeItem) : Boolean = hasScope(sc) || hasScope(sc + ".readonly")
+
+  def oauthScope(scope: ScopeItem): Directive0 = {
+    if (hasScope(scope)) pass
+    else reject(InvalidScopeRejection(scope), AuthorizationFailedRejection)
+  }
+
+  def oauthScopeReadonly(scope: ScopeItem): Directive0 = {
+    if (hasScopeReadonly(scope)) pass
+    else reject(InvalidScopeRejection(scope), AuthorizationFailedRejection)
+  }
+}
+
+object AuthedNamespaceScope {
+  import scala.language.implicitConversions
+  implicit def toNamespace(ns: AuthedNamespaceScope) = ns.namespace
+
+  val namespacePrefix = "namespace."
+
+  def apply(ns: Namespace) : AuthedNamespaceScope = AuthedNamespaceScope(ns, Scope(Set.empty), true)
+
+  def apply(token: IdToken) : AuthedNamespaceScope = {
+    AuthedNamespaceScope(Namespace(token.sub.underlying))
+  }
+
+  def apply(token: JsonWebToken) : AuthedNamespaceScope = {
+    val nsSet = token.scope.underlying.collect {
+      case x if x.startsWith(namespacePrefix) => x.substring(namespacePrefix.length)
+    }
+    if (nsSet.size == 1) {
+      AuthedNamespaceScope(Namespace(nsSet.toVector(0)), token.scope, false)
```

Comment:
I get linting errors for both boolean args here and on line 53, as they are unnamed.